v2/contribute/decisions/oauth/0003-consent-screen.mdx

```diff
@@ -0,0 +1,57 @@
+---
+id: "0003"
+title: No built-in consent screen at first
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# No built-in consent screen at first
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+While not strictly required most OAuth2 implementations include a consent screen.
+
+## Considered Options
+
+* Build consent screen in the first version
+* No built-in consent screen at first
+* No built-in consent screens
+
+## Decision Outcome
+
+Chosen option: No built-in consent screen at first, because
+
+* We can get initial support earlier
+* We can ensure future versions will not need migration
+
+What we need store to enable this in the future:
+- enabled scopes (per client):
+    - required anyway
+- auto allowed scopes (per client):
+    - these scopes are allowed ("consented to") by default for every user, meaning 
```

Comment:
@rishabhpoddar to support this feature we may have to add an additional field (`consent required `)in `client allowed table`..
Should i add it now ?

Reply:
There is already a boolean like `requires_consent` in the `oauth2_client_allowed_scopes` table which will be false for all scopes right now. Later on, when we add the consent screen feature, we can make this boolean modifiable by the dev.